### PR TITLE
fix(kubeapiserver client):allow certificate validation to be skipped

### LINKED ISSUE #8487: helm command with --kube-apiserver --kube-token
Helm version : 3.2.0
Target K8s server version : 1.17
Using Openshift.

I tried to use helm command in my local.

helm list --kubeconfig {kubeconfigFilePath} --kube-context {contextName}
-> It works fine.

helm list --kube-apiserver https://{API server IP}:{API server Port} --kube-token {Kubetoken}
-> It shows error like below

Error: Kubernetes cluster unreachable: Get https://10.251.205.11:6443/version?timeout=32s: x509: certificate signed by unknown authority"

Is there necessary data do I have to add to use "--kube-apiserver" without "--kubeconfig"?

### PR DESCRIPTION
**Pull request by donggangcj**
Close: #8487

Signed-off-by: Dong Gang <dong.gang@daocloud.io>

**What this PR does / why we need it**:
close the issue #8487
**Special notes for your reviewer**:
This PR refers to https://github.com/helm/helm/pull/7230
**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by EarthmanT — APPROVED** [review]

File: cmd/helm/load_plugins.go
Comment (bacongobbler) [inline cmd/helm/load_plugins.go:157]: flags that interact with the kubeconfig should be prefixed with `--kube-`.

Comment (bacongobbler) [inline cmd/helm/load_plugins.go:157]: flags that interact with the kubeconfig should be prefixed with `--kube-`. Please change this to `--kube-insecure-skip-tls-verify`.

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go:53]: Please prefix with `Kube`.

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go:128]: Please write tests for this new function.

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go:158]: kubectl does not accept an environment variable for `--insecure-tls-verify`. Why should Helm?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/cli/environment.go
Comment (donggangcj) [reply pkg/cli/environment.go]: Done;)

File: pkg/cli/environment.go
Comment (donggangcj) [reply pkg/cli/environment.go:128]: Done:)

File: cmd/helm/load_plugins.go
Comment (donggangcj) [reply cmd/helm/load_plugins.go:157]: Done;)

Comment (bacongobbler) [reply cmd/helm/load_plugins.go:157]: It's still `--insecure-skip-tls-verify` here.

**Review by bacongobbler — APPROVED** [review]
LGTM pending one comment.

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Accidentally approved; one code change still needs to occur before merging

**Comment by bacongobbler** [comment]
Also needs a rebase.

**Comment by donggangcj** [comment]
> Also needs a rebase.

Done;)

**Comment by donggangcj** [comment]
@bacongobbler Can this PR be merged into the main branch and is there anything else I need to do？

**Comment by thejasbabu** [comment]
Can we also add the ability to set CAFile from an env variable `HELM_CA_FILE`? Right now, there is no way to pass the CA certificate if one uses the pkg and has the certificate.

**Comment by AdarShaked** [comment]
@bacongobbler  this PR is going to be merged?

**Comment by donggangcj** [comment]
@bacongobbler  Is it possible to merge this pr?

File: pkg/cli/environment.go
Comment (bacongobbler) [inline pkg/cli/environment.go:82]: Why are we being inconsistent with the naming convention here? The feature flag is called `--kube-insecure-skip-tls-verify`, not `--kube-apiserver-insecure`.

Please follow the same convention as similar flags and re-name this as `HELM_KUBEINSECURE_SKIP_TLS_VERIFY`.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/cli/environment.go
Comment (donggangcj) [reply pkg/cli/environment.go]: My bad! the renaming is done now.

File: pkg/cli/environment.go
Comment (donggangcj) [reply pkg/cli/environment.go]: Thanks! Done;

**Comment by donggangcj** [comment]
> As a note,this will require a review from two core maintainers before it can be merged. Even after I review and approve this, you will need to find another core maintainer willing to review the code here.

Okay,  I will request another Matainer to review and open a new PR to improve the certification of helm and kubernetes as #8487 suggested. Finally, Thank you for your patient guidance！

**Comment by donggangcj** [comment]
@hickeyma @technosophos could you please review this pr if you are free?  thanks!

**Comment by AdarShaked** [comment]
Hi, what is the status of this PR?

**Comment by unkis** [comment]
any updates on that?

**Comment by AdarShaked** [comment]
Any updates for that ?

**Comment by koulq** [comment]
@donggangcj  HI ，When I used the parameter kube-insecure-skip-tls-verify, I checked the certificate, kube-insecure-skip-tls-verify didn't work

image

**Comment by koulq** [comment]
@donggangcj  HI ，When I used the parameter kube-insecure-skip-tls-verify, I checked the certificate, kube-insecure-skip-tls-verify didn't work

image

**Comment by davidkarlsen** [comment]
@donggangcj it needs a rebase
@mattfarina wdyt?

**Comment by tuxBurner** [comment]
@davidkarlsen and @donggangcj i rebase it in the pr #9727

**Comment by bacongobbler** [comment]
Closing in favour of https://github.com/helm/helm/pull/10944.

**Pull request closed without merging**
